test/common/nighthawk_sink_client_test.cc

```diff
@@ -0,0 +1,304 @@
+        EXPECT_CALL(*mock_writer, Write(_, _)).WillOnce(Return(false));
+        return mock_writer;
+      });
+
+  NighthawkSinkClientImpl client;
+  absl::StatusOr<nighthawk::client::StoreExecutionResponse> response =
+      client.StoreExecutionResponseStream(&mock_nighthawk_sink_stub, {});
+  ASSERT_FALSE(response.ok());
+  EXPECT_EQ(response.status().code(), absl::StatusCode::kUnavailable);
+  EXPECT_THAT(response.status().message(), HasSubstr("Failed to write"));
+}
+
+TEST(StoreExecutionResponseStream, ReturnsErrorIfNighthawkServiceWritesDoneFails) {
+  nighthawk::client::MockNighthawkSinkStub mock_nighthawk_sink_stub;
+  // Configure the mock Nighthawk Service stub to return an inner mock channel when the code under
+  // test requests a channel. Set call expectations on the inner mock channel.
+  EXPECT_CALL(mock_nighthawk_sink_stub, StoreExecutionResponseStreamRaw)
+      .WillOnce([](grpc::ClientContext*, ::nighthawk::client::StoreExecutionResponse*) {
+        auto* mock_writer = new grpc::testing::MockClientWriter<StoreExecutionRequest>();
+        EXPECT_CALL(*mock_writer, Write(_, _)).WillOnce(Return(true));
+        EXPECT_CALL(*mock_writer, WritesDone()).WillOnce(Return(false));
+        return mock_writer;
+      });
+
+  NighthawkSinkClientImpl client;
+  absl::StatusOr<nighthawk::client::StoreExecutionResponse> response =
+      client.StoreExecutionResponseStream(&mock_nighthawk_sink_stub, {});
+  ASSERT_FALSE(response.ok());
+  EXPECT_EQ(response.status().code(), absl::StatusCode::kInternal);
+  EXPECT_THAT(response.status().message(), HasSubstr("WritesDone() failed"));
+}
+
+TEST(StoreExecutionResponseStream, PropagatesErrorIfNighthawkServiceGrpcStreamClosesAbnormally) {
+  nighthawk::client::MockNighthawkSinkStub mock_nighthawk_sink_stub;
+  // Configure the mock Nighthawk Service stub to return an inner mock channel when the code under
+  // test requests a channel. Set call expectations on the inner mock channel.
+  EXPECT_CALL(mock_nighthawk_sink_stub, StoreExecutionResponseStreamRaw)
+      .WillOnce([](grpc::ClientContext*, ::nighthawk::client::StoreExecutionResponse*) {
+        auto* mock_writer = new grpc::testing::MockClientWriter<StoreExecutionRequest>();
+        EXPECT_CALL(*mock_writer, Write(_, _)).WillOnce(Return(true));
+        EXPECT_CALL(*mock_writer, WritesDone()).WillOnce(Return(true));
+        EXPECT_CALL(*mock_writer, Finish())
+            .WillOnce(
+                Return(::grpc::Status(::grpc::PERMISSION_DENIED, "Finish failure status message")));
+        return mock_writer;
+      });
+
+  NighthawkSinkClientImpl client;
+  absl::StatusOr<nighthawk::client::StoreExecutionResponse> response =
+      client.StoreExecutionResponseStream(&mock_nighthawk_sink_stub, {});
+  ASSERT_FALSE(response.ok());
+  EXPECT_EQ(response.status().code(), absl::StatusCode::kPermissionDenied);
+  EXPECT_THAT(response.status().message(), HasSubstr("Finish failure status message"));
+}
+
+// ---
```

Comment:
Did we mean to leave this in?

Reply:
Remove in 78dad245c0fd4e3f907296efba461ecaaf9b7e7d